Inferences/BinaryResolution.cpp

```diff
@@ -177,71 +182,25 @@ Clause* BinaryResolution::generateClause(Clause* queryCl, Literal* queryLit, SLQ
     }
   }
 
-  bool synthesis = (env.options->questionAnswering() == Options::QuestionAnsweringMode::SYNTHESIS);
-  Literal* cAnsLit = synthesis ? queryCl->getAnswerLiteral() : nullptr;
```

Comment:
What will happen to QuestionAnsweringMode? Is that dealt with in some other way?

Reply:
okayyy sorry it seems like i deleted this accidentally during merging at some point. I didn't ever see this bit of code conciously before (as it's rather new compared to my changes). I'll look into it now.

Reply:
Okay i fixed it in  BR & Superposition.  @hzzv  could you have a quick look whether this looks right?

Reply:
BR & Superposition work now, thanks!
However, your branch crashes for me on this input: https://github.com/vprover/vampire_benchmarks/blob/master/synthesis/non-recursive/smt2/group-ex3-square_commutativity.smt2
with `--decode lrs+10_3:1_av=off:ep=off:lcm=reverse:qa=synthesis:sp=reverse_arity:urr=on_300`.

The issue seems to be somewhere in URR:
```
vampireMode()
??:?
Saturation::ProvingHelper::runVampireSaturation(Kernel::Problem&, Shell::Options const&)
??:?
Saturation::ProvingHelper::runVampireSaturationImpl(Kernel::Problem&, Shell::Options const&)
??:?
Kernel::MainLoop::run()
??:?
Saturation::SaturationAlgorithm::runImpl()
??:?
Saturation::SaturationAlgorithm::activate(Kernel::Clause*)
??:?
Inferences::CompositeSGI::generateSimplify(Kernel::Clause*)
??:?
Inferences::CompositeGIE::generateClauses(Kernel::Clause*)
??:?
Inferences::URResolution::generateClauses(Kernel::Clause*)
??:?
Inferences::URResolution::processAndGetClauses(Inferences::URResolution::Item*, unsigned int, Lib::List<Kernel::Clause*>*&)
??:?
Inferences::URResolution::processLiteral(Lib::List<Inferences::URResolution::Item*>*&, unsigned int)
??:?
??
??:0
Lib::handleSignal(int)
??:?
Debug::Tracer::printStack(std::ostream&)
??:?
```

Reply:
Great thanks :)

[fixed it](https://github.com/vprover/vampire/pull/508/commits/9ecb46d597fbba1793ce890d6adc84d5269763c7)). I'm pretty sure this bug was already present in master.